File: Groupe2/TP1/src/calculs.py
def main():
    # Lecture de la première valeur
    num1 = int(input("Valeur 1 : \n"))

    # Lecture de l'opérateur
    op = input("Entrez un opérateur : ")

    # Pour les opérateurs autres que ~, demander une deuxième valeur
    if op != '~':
        num2 = int(input("Valeur 2 : \n"))

    # Avec IF
    print("Avec if :")
    if op == '+':
        print(f"{num1} {op} {num2} = {num1 + num2}")
    elif op == '-':
        print(f"{num1} {op} {num2} = {num1 - num2}")
    elif op == '*':
        print(f"{num1} {op} {num2} = {num1 * num2}")
    elif op == '/':
        if num2 != 0:
            print(f"{num1} {op} {num2} = {num1 / num2}")
        else:
            print("Erreur : division par zéro")
    elif op == '%':
        if num2 != 0:
            print(f"{num1} {op} {num2} = {num1 % num2}")
        else:
            print("Erreur : division par zéro")
    elif op == '&':
        print(f"{num1} {op} {num2} = {num1 & num2}")
    elif op == '|':
        print(f"{num1} {op} {num2} = {num1 | num2}")
    elif op == '~':
        print(f"{op}{num1} = {~num1}")
    else:
        print("Opérateur invalide")

    # Avec match
    print("Avec match :")
    match op:
        case '+':
            print(f"{num1} {op} {num2} = {num1 + num2}")
        case '-':
            print(f"{num1} {op} {num2} = {num1 - num2}")
        case '*':
            print(f"{num1} {op} {num2} = {num1 * num2}")
        case '/':
            if num2 != 0:
                print(f"{num1} {op} {num2} = {num1 / num2}")
            else:
                print("Erreur : division par zéro")
        case '%':
            if num2 != 0:
                print(f"{num1} {op} {num2} = {num1 % num2}")
            else:
                print("Erreur : division par zéro")
        case '&':
            print(f"{num1} {op} {num2} = {num1 & num2}")
        case '|':
            print(f"{num1} {op} {num2} = {num1 | num2}")
        case '~':
            print(f"{op}{num1} = {~num1}")
        case _:
            print("Opérateur invalide")


if __name__ == "__main__":
    main()

File: Groupe2/TP1/src/calculs.c
#include <stdio.h>

int main()
{
    int num1;
    int num2;
    char op;

    // Demander les valeurs et l'opérateur
    printf("Valeur 1 : \n");
    scanf("%d", &num1);
    printf("Entrez un opérateur : ");
    scanf(" %c", &op); // Note the space before %c to ignore whitespace

    // Si l'opérateur est un opérateur `unaire`, on ne demande pas la deuxième valeur
    if (op != '~')
    {
        printf("Valeur 2 : \n");
        scanf("%d", &num2);
    }

    // Calculer et afficher le résultat
    switch (op)
    {
    case '+':
        printf("%d %c %d = %d\n", num1, op, num2, num1 + num2);
        break;
    case '-':
        printf("%d %c %d = %d\n", num1, op, num2, num1 - num2);
        break;
    case '*':
        printf("%d %c %d = %d\n", num1, op, num2, num1 * num2);
        break;
    case '/':
        if (num2 != 0)
            printf("%d %c %d = %d\n", num1, op, num2, num1 / num2);
        else
            printf("Erreur : division par zéro\n");
        break;
    case '%':
        if (num2 != 0)
            printf("%d %c %d = %d\n", num1, op, num2, num1 % num2);
        else
            printf("Erreur : division par zéro\n");
        break;
    case '&':
        printf("%d %c %d = %d\n", num1, op, num2, num1 & num2);
        break;
    case '|':
        printf("%d %c %d = %d\n", num1, op, num2, num1 | num2);
        break;
    case '~':
        printf("%c%d = %d\n", op, num1, ~num1);
        break;
    default:
        printf("Opérateur invalide\n");
        break;
    }

    return 0;
}
